tools/clang-tidy-plugin/AlmostNeverAutoCheck.cpp

```diff
@@ -107,6 +108,7 @@ static void CheckDecl( AlmostNeverAutoCheck &Check,
     PrintingPolicy Policy( LangOptions{} );
     Policy.adjustForCPlusPlus();
     std::string TypeStr = AutoTp.getAsString( Policy );
+    const std::string DesugaredTypeStr = DesugaredAutoTp.getAsString( Policy );
```

Comment:
```suggestion

    std::string DesugaredTypeStr;
    // Test stripped type is not null to avoid a crash in
    // QualType::getSplitDesugaredType in clang/lib/AST/Type.cpp
    if( QualifierCollector().strip( AutoTp ) ) {
        QualType DesugaredAutoTp = AutoTp.getDesugaredType( *Result.Context );
        DesugaredTypeStr = DesugaredAutoTp.getAsString( Policy );
    }
```

There was a crash here in the original PR and this seems to fix it.